### PR TITLE
Ciena SDS 

### PR DESCRIPTION
**Pull request by penfold1972**
Please give a short description what your pull request is for
Adding interface and CPU graphs for Ciena Service Delivery Switches that run the LEOS version of firmware.
Just adding a few MIBs and editing the ciena-sds.yaml file to discover the data points to graph.

DO NOT DELETE THE UNDERLYING TEXT

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [ ] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)
- [ ] If my Pull Request does some changes/fixes/enhancements in the WebUI, I have inserted a screenshot of it.

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

### CODE REVIEW

**Comment by murrant** [comment]
Changes look good, just needs test data so it doesn't get broken in the future.

**Comment by penfold1972** [comment]
I don't understand what is going wrong with the collecting data:
`bash-4.2$ ./scripts/save-test-data.php  -o ciena-sds
<SNIP>
Polled in 6.341 seconds
Saved to /opt/librenms/tests/data/ciena-sds.json
Ready for testing!
bash-4.2$ ./scripts/pre-commit.php --db --snmpsim
Use ./lnms dev:check
bash-4.2$ `

Should I upload /opt/librenms/tests/data/ciena-sds.json here?

**Comment by Jellyfrog** [comment]
Run:
1. `./scripts/collect-snmp-data.php -h *YOUR DEVICE HOSTNAME*`
2. `./scripts/save-test-data.php -o ciena-sds`

Then commit both the snmprec and json file

**Comment by penfold1972** [comment]
Does the command append to the existing file (/opt/librenms/tests/data/ciena-sds.json), or does it overwrite it?

I ask because I think I my changes apply to the LEOS version of firmware for the Service Delivery Switch (SDS) and I know none of the existing data/graphs apply to that.

I'm concerned about overwriting the test data for the existing discovery with data that only applies to the model I am building graphs against.

**Comment by penfold1972** [comment]
I think I have the data merged.

**Comment by murrant** [comment]
It looks like the processor index values are too long, you will have to manually set them to a shorter value.

**Comment by penfold1972** [comment]
I don't know how to translate that statement into an action.
What file has the bad data?
How do I find the bad data?
When I find that, would it be obvious what I need to change?
Could you give me an example of the bad data that I could grep for?

File: includes/definitions/discovery/ciena-sds.yaml [hunk 1]
```diff
@@ -8,26 +8,90 @@ modules:
                 num_oid: '.1.3.6.1.4.1.1271.2.1.5.1.2.1.4.5.1.4.{{ $index }}'
                 descr: 'Slot {{ $subindex1 }} CPU'
                 snmp_flags: '-OQUsbe'
+            -
+                oid: wwpLeosSystemCpuUtilization
+                value: wwpLeosSystemCpuUtilizationLast5Seconds
+                num_oid: '.1.3.6.1.4.1.6141.2.60.12.1.11.1.{{ $index }}'
+                descr: 'CPU Utilization 5'
+                index: wwpLeosSystemCpuUtilizationLast5Seconds.{{ $index }}
```
Comment (murrant) [inline includes/definitions/discovery/ciena-sds.yaml:16]: This index is too long for the database :D
Just shorten it up.

File: includes/definitions/discovery/ciena-sds.yaml [hunk 2]
```diff
@@ -8,26 +8,90 @@ modules:
...
+                group: 'CPU Utilization'
+                snmp_flags: '-OQUsbe'
+                #low_limit: wwpLeosSystemCpuUtilizationLast5SecondsMinimum
+                #high_limit: wwpLeosSystemCpuUtilizationLast5SecondsMaximum
+            -
+                oid: wwpLeosSystemCpuUtilization
+                value: wwpLeosSystemCpuUtilizationLast10Seconds
+                num_oid: '.1.3.6.1.4.1.6141.2.60.12.1.11.5.{{ $index }}'
+                descr: 'CPU Utilization 10'
+                index: wwpLeosSystemCpuUtilizationLast10Seconds.{{ $index }}
```
Comment (murrant) [inline includes/definitions/discovery/ciena-sds.yaml:26]: And here

File: includes/definitions/discovery/ciena-sds.yaml [hunk 3]
```diff
@@ -8,26 +8,90 @@ modules:
...
+                group: 'CPU Utilization'
+                #low_limit: wwpLeosSystemCpuUtilizationLast10SecondsMinimum
+                #high_limit: wwpLeosSystemCpuUtilizationLast10SecondsMaximum
+            -
+                oid: wwpLeosSystemCpuUtilization
+                value: wwpLeosSystemCpuUtilizationLast60Seconds
+                num_oid: '.1.3.6.1.4.1.6141.2.60.12.1.11.9.{{ $index }}'
+                descr: 'CPU Utilization 60'
+                index: wwpLeosSystemCpuUtilizationLast60Seconds.{{ $index }}
```
Comment (murrant) [inline includes/definitions/discovery/ciena-sds.yaml:35]: and here

File: includes/definitions/discovery/ciena-sds.yaml (hunk 1 above)
Comment (penfold1972) [reply includes/definitions/discovery/ciena-sds.yaml:16]: Ok, I thought the issue was with the test data. No wonder I was lost.

I guess I just have to change it to the "num_oid" field?

File: includes/definitions/discovery/ciena-sds.yaml [hunk 4]
```diff
@@ -8,26 +8,90 @@ modules:
                 num_oid: '.1.3.6.1.4.1.1271.2.1.5.1.2.1.4.5.1.4.{{ $index }}'
                 descr: 'Slot {{ $subindex1 }} CPU'
                 snmp_flags: '-OQUsbe'
+            -
+                oid: wwpLeosSystemCpuUtilization
+                value: wwpLeosSystemCpuUtilizationLast5Seconds
+                num_oid: '.1.3.6.1.4.1.6141.2.60.12.1.11.1.{{ $index }}'
+                descr: 'CPU Utilization 5'
+                index: cpu5.{{ $index }}
+                group: 'CPU Utilization'
```
Comment (murrant) [inline includes/definitions/discovery/ciena-sds.yaml:17]: Processor module does not support grouping

**Comment by murrant** [comment]
CLA is unhappy because one of the commits is owned by root.  You should be able to rebase and squash that commit to fix the issue.

**Comment by penfold1972** [comment]
I actually tried that and ran into the most confusing situation of multiple conflicts that had nothing to do with my changes.

I’ve used git clone hundreds of times, but my developing environment is VI because I am a network guy that can write scripts and edit files. So this is my first PR and now I understand why Linus said years ago that he named the program “git” because it can be rather stupid. :) This PR has taken hours to submit the changes I wanted to submit “real quick” and I still feel like I did the first time I got into a car to learn how to drive a manual transmission.

Steven Cobb | Comporium
Network Internet Services | Network Engineer III
p. 803.326.7710<tel:803.326.7710>
www.comporium.com<http://www.comporium.com/>
Steven.Cobb@comporium.com<mailto:steven.cobb@comporium.com>

**Comment by Jellyfrog** [comment]
It will be squashed anyway..

**Comment by murrant** [comment]
Glad you stuck with it :)

**Comment by murrant** [comment]
This pull request has been mentioned on **LibreNMS Community**. There might be relevant details there:

https://community.librenms.org/t/v1-66-release-changelog-july-2020/12953/1

**Pull request merged**